src/main/java/com/traveltime/sdk/dto/requests/proto/Transportation.java

```diff
@@ -1,35 +1,131 @@
 package com.traveltime.sdk.dto.requests.proto;
 
+import com.igeolise.traveltime.rabbitmq.requests.RequestsCommon;
 import lombok.AllArgsConstructor;
+import lombok.Builder;
 import lombok.Getter;
+import lombok.Value;
 
 public interface Transportation {
-    String getValue();
-    Integer getCode();
+    TransportationType getType();
+    RequestsCommon.Transportation getProtoMessage();
 
+    class Modes {
+        public final static Transportation PUBLIC_TRANSPORT = PublicTransport.builder().build();
+        public final static Transportation DRIVING_AND_PUBLIC_TRANSPORT = DrivingAndPublicTransport.builder().build();
+        public final static Transportation WALKING_FERRY = new TransportationWithoutDetails(TransportationType.Types.WALKING_FERRY);
+        public final static Transportation CYCLING_FERRY = new TransportationWithoutDetails(TransportationType.Types.CYCLING_FERRY);
+        public final static Transportation DRIVING_FERRY = new TransportationWithoutDetails(TransportationType.Types.DRIVING_FERRY);
+        public final static Transportation WALKING = new TransportationWithoutDetails(TransportationType.Types.WALKING);
+        public final static Transportation CYCLING = new TransportationWithoutDetails(TransportationType.Types.CYCLING);
+        public final static Transportation DRIVING = new TransportationWithoutDetails(TransportationType.Types.DRIVING);
+    }
+
+    @Builder
```

Comment:
I think we need to make this into a `@With` not `@Builder` if we're going to go with https://github.com/traveltime-dev/traveltime-sdk-java/pull/169#discussion_r2073824059